tools/afl-fuzzer/platform/byt-host.c

```diff
@@ -437,23 +437,44 @@ static int byt_platform_init(struct fuzz *fuzzer,
 	/* initialise bridge to qemu */
 	qemu_io_register_parent(platform->name, &bridge_cb, (void *)fuzzer);
 
-	/* set boot wait timeout */
-	gettimeofday(&tp, NULL);
-	timeout.tv_sec  = tp.tv_sec;
-	timeout.tv_nsec = tp.tv_usec * 1000;
-	timeout.tv_sec += 5;
+	if (to_boot == 'y') {
+		/* set boot wait timeout */
+		gettimeofday(&tp, NULL);
+		timeout.tv_sec  = tp.tv_sec;
+		timeout.tv_nsec = tp.tv_usec * 1000;
+		timeout.tv_sec += 5;
 
-	/* first lock the boot wait mutex */
-	pthread_mutex_lock(&data->mutex);
+		/* first lock the boot wait mutex */
+		pthread_mutex_lock(&data->mutex);
 
-	/* now wait for mutex to be unlocked by boot ready message */
-	while (!ret && !data->boot_complete)
-		ret = pthread_cond_timedwait(&cond, &data->mutex, &timeout);
+		/* now wait for mutex to be unlocked by boot ready message */
+		while (!ret && !data->boot_complete)
+			ret = pthread_cond_timedwait(&cond, &data->mutex, &timeout);
 
-	if (ret == ETIMEDOUT && !data->boot_complete)
-		fprintf(stderr, "error: DSP boot timeout\n");
+		if (ret == ETIMEDOUT && !data->boot_complete)
+			fprintf(stderr, "error: DSP boot timeout\n");
 
-	pthread_mutex_unlock(&data->mutex);
+		pthread_mutex_unlock(&data->mutex);
+	} else {
+		// If DSP is already booted, initialize the mailbox offsets
+
+		struct byt_data *data = fuzzer->platform_data;
+
+		data->host_box.offset = 0x400;
+		data->host_box.size = 0x400;
+		data->dsp_box.offset = 0;
+		data->dsp_box.size = 0x400;
```

Comment:
Can we read this from ready message ?

Reply:
No.

It goes something like this: We start the DSP, DSP sends an FW ready message. The first platform which tries to connect to the DSP gets the FW ready message and initializes offsets. We do some work and we the free the platform. Now, when we try to connect second platform to the DSP, it doesn't get the FW ready message because we didn't start/restart the DSP. So, we need some mechanism to set the mailbox offsets and this logic here helps in doing that.

Reply:
@lgirdwood this comes straight from the sof-fuzzer code, i think we can build a platform library for linking into the fuzzers to avoid code duplication.